design/user_data_incentives.md

```diff
@@ -0,0 +1,124 @@
+4. Event query or subscription - used by dApps to update the UIs 
+
+
+## Chains with privacy
+
+It gets a bit more complicated when privacy is added to the mix, because the results of the user interactions are dependent on the requester.
+Only logic run inside an approved TEE is able to return a result, since it is the only place where that data is visible.
+
+This means that it is now much more difficult to create a cheap caching layer.
+Also, the service provider will not gain any insights from this service since all traffic and processing is encrypted.
+
+
+## Delegate to the nodes
+
+The usual rescue in the crypto space is to employ incentives.
+
+Obscuro is a decentralised network of nodes with different roles who already have their own incentives (Aggregators and Verifiers).
+
+The ideal scenario is to have a large and diverse community of verifiers node to make sure that the network functions correctly.
+
+Is is natural to assign to the verifier nodes the additional task of servicing user requests.
+
+### Incentives
+
+There has to be a reason for a node operator to provide this service to users. If there is no payoff, then most operators could just block the ports
+and not accept incoming connections.
+
+Here is a list and analyisis of various potential reasons for offering this service
+
+#### 1. App developer
+
+Someone developed a dApp and wants to support its users, since they are directly interested in the growth of the application. It could be a company, or a DAO.
+E.g.: A game developer would not want to reply to requests made by a DEX user.
+
+For this to be feasible, the enclave must be able to restrict requests to a whitelist of apps. Which is a technical feature that has to be implemented. 
+Also the wallet extension (tooling) needs to know which node to use for which call.
+
+
+#### 2. Fees
+
+Node operators could charge fees for this service.
+
+Given that everyone is now expecting this to be a free service, this is unlikely to be something that has a chance.
+
+
+#### 3. Incentives payed by the protocol.
+
+The network (or protocol) charges fees from user when submitting transactions. This is something that users expect to pay.
+
+The Obscuro protocol is designed in such a way that it decouples the income from the costs by maintaing a buffer.
+
+We can use this designed mechanism to pay for node usage as well along with the L1 gas fees and the general incentives to follow the protocol.
+
+
+##### Measuring node usage
+
+As a proxy for a node responding to user requests, we can use a model where a node is payed a percentage of the gas fees that originated from their node. 
+
+A user that is connected to a node that doesn't respond to requests (like transaction receipts, or events), will leave that node and connect to a different node.
+
+The "Wallet extension", our client-side software can hide this complexity from the user, and switch nodes without the user noticing.
```

Comment:
Should the wallet extension switch a user to a node based on their geographical location too so a user is not traversing from Asia to US for their requests?

Reply:
yeah, that's definitely a valid rule for routing. Proximity